_posts/2024-09-12-opensearch-project-roadmap-2024-2025.md

```diff
@@ -0,0 +1,187 @@
+
+[**Data Prepper pipeline DLQ**](https://github.com/opensearch-project/data-prepper/issues/3857): Data Prepper provides resilience when OpenSearch is down by buffering data and eventually writing to a dead-letter queue (DLQ) if the cluster remains unavailable. Currently, the supported DLQ targets are local file and Amazon S3. One current limitation is that data only goes into the DLQ if it fails to write to the sink. Other failures, such as during processing in the pipeline, do not send to the DLQ. With the proposed pipeline DLQ, Data Prepper will be able to send failed events to the pipeline DLQ or continue to send them downstream, allowing the pipeline author to decide. This will improve the resiliency of data throughout the pipeline. Additionally, the pipeline DLQ will be a pipeline just like any other and will be able to write to any supported Data Prepper sink, such as Apache Kafka. 
```

Comment:
```suggestion
[**Data Prepper pipeline DLQ**](https://github.com/opensearch-project/data-prepper/issues/3857): Data Prepper provides resilience when OpenSearch is down by buffering data and eventually writing to a dead-letter queue (DLQ) if the cluster remains unavailable. Currently supported DLQ targets are local files and Amazon S3. One current limitation is that data is only sent to the DLQ if it fails to write to the sink. Other failures, such as during processing in the pipeline, do not case data to be sent to the DLQ. With the proposed pipeline DLQ, Data Prepper will be able to send failed events to the DLQ or continue to send them downstream, allowing the pipeline author to decide. This will improve the resiliency of data throughout the pipeline. Additionally, the pipeline DLQ will be a pipeline just like any other and will be able to write to any supported Data Prepper sink, such as Apache Kafka. 
```